rust/kernel/amba.rs

```diff
@@ -0,0 +1,268 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Amba devices drivers.
+//!
+//! C header: [`include/linux/amba/bus.h`](../../../../include/linux/amba/bus.h)
+
+use crate::{
+    bindings, c_types, driver, from_kernel_result, io_mem::Resource, power, str::CStr, to_result,
+    types::PointerWrapper, Error, Result,
+};
+use core::{marker::PhantomData, ops::Deref};
+
+/// A registration of an amba driver.
+pub type Registration<T> = driver::Registration<Adapter<T>>;
+
+/// ID of an Amba device.
```

Comment:
Nit: we are using `id`, `ID`, etc. Perhaps we could use `ID` for "prose" and

    `id`

to refer to arguments?

Reply:
Now that I think about, I don't think I like "ID" because it's not an acronym. I replaced all usages with just lower case "id", except when it's the first word of a sentence, in which case I use "Id".